### DISCUSSION TITLE
Ubuntu - GUI

### DISCUSSION THREAD

**Discussion by toddbanister** [pawelsalawa/sqlitestudio#4802] [General]
I compiled everything correctly and without issue (as far as I know) but the output folder only had the program for the CLI (which works). The GUI directory doesn't have any files in it. Any idea what I did wrong?

OS - Ubuntu 22.04 ARM 64-bit

Thanks,
TB

**Comment by pawelsalawa** [comment]
So, what files do you have in the output directory? Can you paste the directory listing?

**Reply by toddbanister** [reply]
Within the Build directory, the Plugins and the guiSQLiteStudio folders are both empty.

build/coreSQLiteStudio Folder:
BigInt.o
Key.o
KeyPair.o
Makefile
PrimeGenerator.o
RSA.o
StackWalker.o
abstractdb.o
asyncqueryrunner.o
attachguard.o
bistrhash.o
blockingsocket.o
blockingsocketprivate.o
builtinplugin.o
cfgcategory.o
cfgentry.o
cfglazyinitializer.o
cfgmain.o
chainexecutor.o
chillout.o
codeformatter.o
codesnippetmanager.o
collationmanagerimpl.o
column.o
committable.o
common.o
compatibility.o
completioncomparer.o
completionhelper.o
config.o
configimpl.o
csvformat.o
csvserializer.o
datatype.o
db.o
dbattacher.o
dbattacherimpl.o
dbmanager.o
dbmanagerimpl.o
dbobjectorganizer.o
dbpluginsqlite3.o
dbpluginstdfilebase.o
dbsqlite3.o
ddlhistorymodel.o
diff_match_patch.o
expectedtoken.o
exportmanager.o
exportworker.o
extralicensemanager.o
functionmanager.o
functionmanagerimpl.o
genericexportplugin.o
genericplugin.o
importmanager.o
importworker.o
invaliddb.o
keywords.o
lazytrigger.o
lexer.o
lexer_low_lev.o
log.o
memoryusage.o
moc_abstractdb.cpp
moc_abstractdb.o
moc_asyncqueryrunner.cpp
moc_asyncqueryrunner.o
moc_blockingsocket.cpp
moc_blockingsocket.o
moc_blockingsocketprivate.cpp
moc_blockingsocketprivate.o
moc_builtinplugin.cpp
moc_builtinplugin.o
moc_cfgcategory.cpp
moc_cfgcategory.o
moc_cfgentry.cpp
moc_cfgentry.o
moc_chainexecutor.cpp
moc_chainexecutor.o
moc_codesnippetmanager.cpp
moc_codesnippetmanager.o
moc_collationmanager.cpp
moc_collationmanager.o
moc_completionhelper.cpp
moc_completionhelper.o
moc_config.cpp
moc_config.o
moc_configimpl.cpp
moc_configimpl.o
moc_datatype.cpp
moc_datatype.o
moc_db.cpp
moc_db.o
moc_dbmanager.cpp
moc_dbmanager.o
moc_dbmanagerimpl.cpp
moc_dbmanagerimpl.o
moc_dbobjectorganizer.cpp
moc_dbobjectorganizer.o
moc_dbpluginsqlite3.cpp
moc_dbpluginsqlite3.o
moc_ddlhistorymodel.cpp
moc_ddlhistorymodel.o
moc_exportmanager.cpp
moc_exportmanager.o
moc_exportworker.cpp
moc_exportworker.o
moc_functionmanager.cpp
moc_functionmanager.o
moc_functionmanagerimpl.cpp
moc_functionmanagerimpl.o
moc_genericexportplugin.cpp
moc_genericexportplugin.o
moc_genericplugin.cpp
moc_genericplugin.o
moc_importmanager.cpp
moc_importmanager.o
moc_importworker.cpp
moc_importworker.o
moc_lazytrigger.cpp
moc_lazytrigger.o
moc_notifymanager.cpp
moc_notifymanager.o
moc_pluginmanager.cpp
moc_pluginmanager.o
moc_pluginmanagerimpl.cpp
moc_pluginmanagerimpl.o
moc_pluginservicebase.cpp
moc_pluginservicebase.o
moc_populateconstant.cpp
moc_populateconstant.o
moc_populatedictionary.cpp
moc_populatedictionary.o
moc_populatemanager.cpp
moc_populatemanager.o
moc_populaterandom.cpp
moc_populaterandom.o
moc_populaterandomtext.cpp
moc_populaterandomtext.o
moc_populatescript.cpp
moc_populatescript.o
moc_populatesequence.cpp
moc_populatesequence.o
moc_populateworker.cpp
moc_populateworker.o
moc_predefs.h
moc_queryexecutor.cpp
moc_queryexecutor.o
moc_queryexecutoraddrowids.cpp
moc_queryexecutoraddrowids.o
moc_queryexecutorattaches.cpp
moc_queryexecutorattaches.o
moc_queryexecutorcolumns.cpp
moc_queryexecutorcolumns.o
moc_queryexecutorcolumntype.cpp
moc_queryexecutorcolumntype.o
moc_queryexecutorcountresults.cpp
moc_queryexecutorcountresults.o
moc_queryexecutordatasources.cpp
moc_queryexecutordatasources.o
moc_queryexecutordetectschemaalter.cpp
moc_queryexecutordetectschemaalter.o
moc_queryexecutorexecute.cpp
moc_queryexecutorexecute.o
moc_queryexecutorexplainmode.cpp
moc_queryexecutorexplainmode.o
moc_queryexecutorfilter.cpp
moc_queryexecutorfilter.o
moc_queryexecutorlimit.cpp
moc_queryexecutorlimit.o
moc_queryexecutororder.cpp
moc_queryexecutororder.o
moc_queryexecutorparsequery.cpp
moc_queryexecutorparsequery.o
moc_queryexecutorreplaceviews.cpp
moc_queryexecutorreplaceviews.o
moc_queryexecutorstep.cpp
moc_queryexecutorstep.o
moc_queryexecutorvaluesmode.cpp
moc_queryexecutorvaluesmode.o
moc_queryexecutorwrapdistinctresults.cpp
moc_queryexecutorwrapdistinctresults.o
moc_querymodel.cpp
moc_querymodel.o
moc_scriptingqt.cpp
moc_scriptingqt.o
moc_scriptingqtdbproxy.cpp
moc_scriptingqtdbproxy.o
moc_scriptingsql.cpp
moc_scriptingsql.o
moc_signalwait.cpp
moc_signalwait.o
moc_sqlfileexecutor.cpp
moc_sqlfileexecutor.o
moc_sqlhistorymodel.cpp
moc_sqlhistorymodel.o
moc_sqliteextensionmanager.cpp
moc_sqliteextensionmanager.o
moc_sqlitestatement.cpp
moc_sqlitestatement.o
moc_sqlitestudio.cpp
moc_sqlitestudio.o
moc_threadwitheventloop.cpp
moc_threadwitheventloop.o
moc_updatemanager.cpp
moc_updatemanager.o
notifymanager.o
nulldevice.o
parser.o
parser_helper_stubs.o
parsercontext.o
parsererror.o
pluginmanagerimpl.o
pluginservicebase.o
pluginsymbolresolver.o
plugintype.o
populateconstant.o
populatedictionary.o
populatemanager.o
populaterandom.o
populaterandomtext.o
populatescript.o
populatesequence.o
populateworker.o
posixcrashhandler.o
qio.o
qmake_qmake_qm_files.qrc
qrc_coreSQLiteStudio.cpp
qrc_coreSQLiteStudio.o
qrc_qmake_qmake_qm_files.cpp
qrc_qmake_qmake_qm_files.o
queryexecutor.o
queryexecutoraddrowids.o
queryexecutorattaches.o
queryexecutorcolumns.o
queryexecutorcolumntype.o
queryexecutorcountresults.o
queryexecutordatasources.o
queryexecutordetectschemaalter.o
queryexecutorexecute.o
queryexecutorexplainmode.o
queryexecutorfilter.o
queryexecutorlimit.o
queryexecutororder.o
queryexecutorparsequery.o
queryexecutorreplaceviews.o
queryexecutorstep.o
queryexecutorvaluesmode.o
queryexecutorwrapdistinctresults.o
querygenerator.o
querymodel.o
readwritelocker.o
returncode.o
schemaresolver.o
scriptingqt.o
scriptingqtdbproxy.o
scriptingsql.o
selectresolver.o
signalwait.o
sqlerrorcodes.o
sqlerrorresults.o
sqlfileexecutor.o
sqlformatterplugin.o
sqlhistorymodel.o
sqlite3_parse.o
sqlitealtertable.o
sqliteanalyze.o
sqliteattach.o
sqlitebegintrans.o
sqlitecolumntype.o
sqlitecommittrans.o
sqliteconflictalgo.o
sqlitecreateindex.o
sqlitecreatetable.o
sqlitecreatetrigger.o
sqlitecreateview.o
sqlitecreatevirtualtable.o
sqlitedeferrable.o
sqlitedelete.o
sqlitedetach.o
sqlitedropindex.o
sqlitedroptable.o
sqlitedroptrigger.o
sqlitedropview.o
sqliteemptyquery.o
sqliteexpr.o
sqliteextensionmanagerimpl.o
sqlitefilterover.o
sqliteforeignkey.o
sqliteindexedcolumn.o
sqliteinsert.o
sqlitelimit.o
sqlitenulls.o
sqliteorderby.o
sqlitepragma.o
sqlitequery.o
sqlitequerytype.o
sqliteraise.o
sqlitereindex.o
sqliterelease.o
sqliterollback.o
sqlitesavepoint.o
sqliteselect.o
sqlitesortorder.o
sqlitestatement.o
sqlitestudio.o
sqliteupdate.o
sqliteupsert.o
sqlitevacuum.o
sqlitewindowdefinition.o
sqlitewith.o
sqlquery.o
sqlresultsrow.o
statementtokenbuilder.o
table.o
tablemodifier.o
threadwitheventloop.o
token.o
translations.o
tsvserializer.o
updatemanager.o
utils.o
utils_sql.o
viewmodifier.o
windowscrashhandler.o
xmldeserializer.o


build/sqlitestudiocli Folder:
Makefile               climsghandler.o             moc_clicommandhistory.o
cli.o                  cliutils.o                  moc_clicommandmode.cpp
cli_config.o           main.o                      moc_clicommandmode.o
clicommand.o           moc_cli.cpp                 moc_clicommandnullvalue.cpp
clicommandadd.o        moc_cli.o                   moc_clicommandnullvalue.o
clicommandcd.o         moc_clicommand.cpp          moc_clicommandopen.cpp
clicommandclose.o      moc_clicommand.o            moc_clicommandopen.o
clicommanddblist.o     moc_clicommandadd.cpp       moc_clicommandpwd.cpp
clicommanddesc.o       moc_clicommandadd.o         moc_clicommandpwd.o
clicommanddir.o        moc_clicommandcd.cpp        moc_clicommandremove.cpp
clicommandexecutor.o   moc_clicommandcd.o          moc_clicommandremove.o
clicommandexit.o       moc_clicommandclose.cpp     moc_clicommandsql.cpp
clicommandfactory.o    moc_clicommandclose.o       moc_clicommandsql.o
clicommandhelp.o       moc_clicommanddblist.cpp    moc_clicommandtables.cpp
clicommandhistory.o    moc_clicommanddblist.o      moc_clicommandtables.o
clicommandmode.o       moc_clicommanddesc.cpp      moc_clicommandtree.cpp
clicommandnullvalue.o  moc_clicommanddesc.o        moc_clicommandtree.o
clicommandopen.o       moc_clicommanddir.cpp       moc_clicommanduse.cpp
clicommandpwd.o        moc_clicommanddir.o         moc_clicommanduse.o
clicommandremove.o     moc_clicommandexecutor.cpp  moc_predefs.h
clicommandsql.o        moc_clicommandexecutor.o    qmake_qmake_qm_files.qrc
clicommandsyntax.o     moc_clicommandexit.cpp      qrc_qmake_qmake_qm_files.cpp
clicommandtables.o     moc_clicommandexit.o        qrc_qmake_qmake_qm_files.o
clicommandtree.o       moc_clicommandhelp.cpp      qrc_sqlitestudiocli.cpp
clicommanduse.o        moc_clicommandhelp.o        qrc_sqlitestudiocli.o
clicompleter.o         moc_clicommandhistory.cpp

output/SQLiteStudio Folder:
sqlitestudiocli

output/SQLiteStudio/lib Folder:
libcoreSQLiteStudio.so
libcoreSQLiteStudio.so.1
libcoreSQLiteStudio.so.1.0
libcoreSQLiteStudio.so.1.0.0

Let me know if you need anything else.

Take care,
TB

**Reply by pawelsalawa** [reply]
Can you tell me how did you compile the app? From which folder did you execute the `qmake` and `make`? And with what parameters?

**Reply by toddbanister** [reply]
I simply ran the compile.sh from the /scripts/linux directory with no modifications.

**Reply by pawelsalawa** [reply]
Hm. Maybe there was an error, but you missed it, because the script continued with `Plugins` subdirectory.

Try this: go to `output/build` and execute `make`. See if there are any errors in the end.

**Reply by toddbanister** [reply]
parallels@ubuntu-linux-22-04-desktop:~/Downloads/sqlitestudio-3.4.4/output/build$ make

cd coreSQLiteStudio/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /home/parallels/Downloads/sqlitestudio-3.4.4/SQLiteStudio3/coreSQLiteStudio/coreSQLiteStudio.pro CONFIG+=portable ) && make -f Makefile 
make[1]: Entering directory '/home/parallels/Downloads/sqlitestudio-3.4.4/output/build/coreSQLiteStudio'
make[1]: Nothing to be done for 'first'.
make[1]: Leaving directory '/home/parallels/Downloads/sqlitestudio-3.4.4/output/build/coreSQLiteStudio'
cd guiSQLiteStudio/ && ( test -e Makefile || /usr/lib/qt5/bin/qmake -o Makefile /home/parallels/Downloads/sqlitestudio-3.4.4/SQLiteStudio3/guiSQLiteStudio/guiSQLiteStudio.pro CONFIG+=portable ) && make -f Makefile 
Project ERROR: Unknown module(s) in QT: uitools svg
make: *** [Makefile:75: sub-guiSQLiteStudio-make_first] Error 3

**Reply by pawelsalawa** [reply]
Following these answers:
https://stackoverflow.com/questions/21098805/unknown-modules-in-qt-svg
https://stackoverflow.com/questions/20670457/qt-5-unknown-modules-in-qt-uitools
you probably need to:
```
sudo apt-get install libqt5svg5*
sudo apt-get install qttools5-dev
```
The second package may be named also `libqt5designer5` - according to answers from StackOverflow.
Anyway, you need to install `uitools` and `svg` modules of your Qt installation and then retry the `make`.

**Reply by toddbanister** [reply]
That was it! 

I was able to run the compile.sh again and everything worked as expected. I was able to now to open up the sqlitestudio gui without any issue. Thank you SO very much for the assistance.

Take care,
TB
